### ISSUE TITLE
linter: `react/jsx-no-constructed-context-values` misses imported React 19 context providers

### ISSUE THREAD

**Issue by lukpsaxo** [oxc-project/oxc#26575]
### What version of Oxlint are you using?

1.82.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
    "$schema": "./node_modules/oxlint/configuration_schema.json",
    "plugins": ["react"],
    "rules": {
        "react/jsx-no-constructed-context-values": "error"
    }
}
```

### What happened?

`context.ts`:

```tsx
import { createContext } from 'react';

export const ImportedContext = createContext<unknown>(null);
```

`repro.tsx`:

```tsx
import { ImportedContext } from './context';

export function Repro() {
    return (
        <>
            <ImportedContext value={{}} />
            <ImportedContext.Provider value={{}} />
        </>
    );
}
```

## Actual result

Only `<ImportedContext.Provider value={{}} />` is reported.

## Expected result

Both constructed values are reported. In React 19, `<ImportedContext value={...}>` is the provider
form of `<ImportedContext.Provider value={...}>`.

## Additional context

Moving `createContext` into `repro.tsx` makes Oxlint report the shorthand usage. This indicates the
value-expression analysis works, but context recognition does not follow an imported context for
React 19 shorthand.

**Comment by camc314** [comment]
Hi, thanks for reporting this issue.

I'm looking into this now, however it appears we have the same behaviour as the upstream eslint-plugin-react:
| Context declaration | `<Context value={{}} />` | `<Context.Provider value={{}} />` |
|---|---|---|
| Imported | Not reported | Reported |
| Local `createContext()` | Reported | Reported |

Its [implementation](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/lib/rules/jsx-no-constructed-context-values.js) recognizes shorthand only through a local `createContext()` initializer; `.Provider` is recognized by name.

WIthout type information, i'm not sure it's possible to detect this usage.

**Comment by lukpsaxo** [comment]
I am using eslint plugin react x since the original react plugin is so badly mantained. It uses a heuristic on the context name, which I agree is less reliable than typing or detecting createContext locally, but it works for us.

Since the original rule doesn't catch this I am happy to close if you don't want to use a name heuristic.

source:
https://github.com/Rel1cx/eslint-react/blob/main/plugins/eslint-plugin-react-x/src/rules/no-unstable-context-value/lib.ts#L7
